app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt

```diff
@@ -43,8 +44,19 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
                 it.value.contentLength.toString(),
                 it.value.contentType
             )
+        }.toMutableList()
+
+        val notFoundDownloads = emptyList<DownloadItem>().toMutableList()
+        for (item in items) {
+            val filePath = item.filePath
+            val file = File(filePath)
+            if (!file.exists()) {
+                notFoundDownloads.add(item)
+            }
         }
```

Comment:
Yep! That is a good idea, I just didn't know if this could be done in a better way or if it's in the right place, maybe I need to change it? @Amejia481

Reply:
I think this code is sufficient, as this is logic, we could have it separate place a test it in isolation. Maybe an [extension function](https://kotlinlang.org/docs/reference/extensions.html) something like `DownloadState.existsOnDisk` or something similar

Reply:
Is this what you had in mind? I assume I would then add a test to the 'ext' test folder where all the other extension tests live?